matRad_calcParticleDoseMCtopas.m

```diff
@@ -158,58 +183,130 @@
                 
                 %Overwrite CT (TEMPORARY - we should use 4D calculation in
                 %TOPAS here)
-                ctR.cubeHU = cubeHUresampled(ctScen);
-                ctR.cube = cubeResampled(ctScen);
-                                
-                topasConfig.writeAllFiles(ctR,pln,stf,topasBaseData,w);                
+%                 ctR.cubeHU = cubeHUresampled(ctScen);
+%                 ctR.cube = cubeResampled(ctScen);
+                %Delete previous topas files
+                files = dir([topasConfig.workingDir,'*']);
+                files = {files(~[files.isdir]).name};
+                fclose('all');
+                for i = 1:length(files)
+                    delete([topasConfig.workingDir,files{i}])
+                end
+               
+                if calcDoseDirect
+                    topasConfig.writeAllFiles(ctR,pln,stf,topasBaseData,w(:,ctScen));
+                else
+                    topasConfig.writeAllFiles(ctR,pln,stf,topasBaseData);
+                end
                 
                 % Run simulation for current scenario
                 cd(topasConfig.workingDir);
-                
-                for beamIx = 1:numel(stf)
-                    for runIx = 1:topasConfig.numOfRuns       
-                        fname = sprintf('%s_field%d_run%d',topasConfig.label,beamIx,runIx);
-                        topasCall = sprintf('%s %s.txt > %s.out 2> %s.err',topasConfig.topasExecCommand,fname,fname,fname);
-                        if topasConfig.parallelRuns
-                            finishedFiles{runIx} = sprintf('%s.finished',fname);
-                            delete(finishedFiles{runIx});
-                            topasCall = [topasCall '; touch ' finishedFiles{runIx} ' &'];
+                              
+                if exportForExternalCalculation
```

Comment:
Yes, in order to read the calculated files back into matRad, meta information about e.g. the dij is needed. That information is now selectively stored in the MCparam struct.
I don't know exactly what you mean with providing a function handle, since the workflows to run topas externally are so vastly different?

Reply:
let's say you specify an interface for starting the simulation and gathering the files. you can say that the starting function would always needs to be a function that gets the parameters (topasFiles, folder, blablabla). Then someone could write a function getting this parameters to launch the simulation somewhere, or call some other script etc., and just provide the handle of this function: `fHandle = @myStartupFunction` and then our code calls `fHandle(topasFiles, folder, blablabla)`.

You don't need to implement this right now, but you could write a TODO comment. It would be a nice feature.